.github/workflows/scripts/build_helper.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+#
+# build_helper.sh
+# This script are used by the CI/CD builds with Github Actions workflow.
+# It gathers some info on software versions and saves it to file 
+# build_helper.sh.out which are later used by the actions workflow.
+# It also patches Makefiles to support building under Github Action
+# runners environment and compile kernels for every compute capability
+# device supported by the NVCC.
+#
+# Copyright (c) 2025 NStorm (https://github.com/N-Storm/)
+# This file is part of mfaktc.
+# Copyright (C) 2009, 2010, 2011  Oliver Weihe (o.weihe@t-online.de)
+#
+# mfaktc is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# mfaktc is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with mfaktc.  If not, see <http://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+if [[ -z "$1" ]]; then
+  echo "Usage: $0 <CUDA version>" >&2
+  exit 1
+fi
+
+# Windows may have it's sort first on PATH, so set this var to the full path 
+# to GNU sort to avoid having writing full path every time.
+export GSORT='/usr/bin/sort'
+
+declare -a CUDA_VERSION
+CUDA_VERSION=( $(echo "$1" | head -n1 | grep -Eom1 -e '^[1-9]([0-9])?\.[0-9]{1,2}(\.[0-9]{1,3})?$' | tr '.' ' ') )
+if [[ -z "${CUDA_VERSION[*]}" ]]; then
+  echo "ERROR! Can't parse CUDA version $1" >&2
+  exit 2
+fi
+
+eval $(echo -e "CUDA_VER_MAJOR=${CUDA_VERSION[0]}\nCUDA_VER_MINOR=${CUDA_VERSION[1]}" | tee $0.out)
```

Comment:
```suggestion
CUDA_VER_MAJOR=${CUDA_VERSION[0]}
CUDA_VER_MINOR=${CUDA_VERSION[1]}
echo -e "CUDA_VER_MAJOR=${CUDA_VERSION[0]}\nCUDA_VER_MINOR=${CUDA_VERSION[1]}" > "$0.out"
```
`eval` is dangerous.